pyquil/reference_simulator.py

```diff
@@ -165,6 +165,27 @@ def __init__(self, n_qubits: int, rs: RandomState = None):
         self.rs = rs
         self.density = np.zeros((2 ** n_qubits, 2 ** n_qubits), dtype=np.complex128)
         self.density[0, 0] = complex(1.0, 0)
+        self.initial_density = None
+
+    def set_density(self, state_matrix):
+        """
+        The default state of ReferenceDensitySimulator is |000...00>. This method is the correct
+        way (TM) to set the state matrix to another state.
+
+        To restore the ReferenceDensitySimulator state back to the default behavior set
+        state_matrix = None.
+
+        :param state_matrix: numpy.ndarray
+        :return: None
+        """
+        rows, cols = state_matrix.shape
+        if rows != cols:
+            raise ValueError("The state matrix is not square.")
+        if self.n_qubits != int(np.log2(rows)):
+            raise ValueError("The state matrix is not defined on the same numbers of qubits as "
+                             "the QVM.")
+        self.density = state_matrix
+        self.initial_density = state_matrix
```

Comment:
Maybe follow the style of the other class functions and `return self`.

Reply:
- [x]  todo